tests/test_exporter.py

```diff
@@ -1804,6 +1804,110 @@ def test_exporter_exports_requirements_txt_with_legacy_packages_and_duplicate_so
     assert content == expected
 
 
+def test_exporter_exports_requirements_txt_with_default_and_secondary_sources(
+    tmp_dir: str, poetry: Poetry
+) -> None:
+    poetry.pool.remove_repository("PyPI")
+    poetry.config.merge(
+        {
+            "repositories": {
+                "custom-a": {"url": "https://a.example.com/simple"},
+                "custom-b": {"url": "https://b.example.com/simple"},
+            },
+            "http-basic": {
+                "custom-a": {"username": "foo", "password": "bar"},
+                "custom-b": {"username": "baz", "password": "qux"},
+            },
+        }
+    )
+    poetry.pool.add_repository(
+        LegacyRepository(
+            "custom-b",
+            "https://b.example.com/simple",
+            config=poetry.config,
+        ),
+        default=True,
+    )
+    poetry.pool.add_repository(
+        LegacyRepository(
+            "custom-a",
+            "https://a.example.com/simple",
+            config=poetry.config,
+        ),
+        secondary=True,
+    )
+    poetry.locker.mock_lock_data(  # type: ignore[attr-defined]
+        {
+            "package": [
+                {
+                    "name": "foo",
+                    "version": "1.2.3",
+                    "category": "main",
+                    "optional": False,
+                    "python-versions": "*",
+                    "source": {
+                        "type": "legacy",
+                        "url": "https://a.example.com/simple",
+                        "reference": "",
+                    },
+                },
+                {
+                    "name": "bar",
+                    "version": "4.5.6",
+                    "category": "dev",
+                    "optional": False,
+                    "python-versions": "*",
+                    "source": {
+                        "type": "legacy",
+                        "url": "https://b.example.com/simple",
+                        "reference": "",
+                    },
+                },
+                {
+                    "name": "baz",
+                    "version": "7.8.9",
+                    "category": "dev",
+                    "optional": False,
+                    "python-versions": "*",
+                    "source": {
+                        "type": "legacy",
+                        "url": "https://b.example.com/simple",
+                        "reference": "",
+                    },
+                },
+            ],
+            "metadata": {
+                "python-versions": "*",
+                "content-hash": "123456789",
+                "hashes": {"foo": ["12345"], "bar": ["67890"], "baz": ["24680"]},
```

Comment:
a recent commit stopped the unit test from using the old `hashes` key, pleae update this one to match